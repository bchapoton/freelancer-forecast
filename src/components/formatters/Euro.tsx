import React, { Fragment } from 'react';
import { Box } from '@mui/material';

export type EuroProps = {
    children?: number;
    cents?: boolean;
    bold?: boolean;
};

function Euro({ children, cents = true, bold = false }: EuroProps) {
    if (!children && children !== 0) return <Fragment />;
    return (
        <Box component="span" sx={{ fontWeight: bold ? 'bold' : 'normal' }}>
            {new Intl.NumberFormat('fr-FR', {
                style: 'currency',
                currency: 'EUR',
                maximumFractionDigits: cents ? 2 : 0,
            }).format(children)}
        </Box>
    );
}

export default Euro;
