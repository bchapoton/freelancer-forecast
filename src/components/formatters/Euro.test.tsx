import React from 'react';
import Euro from './Euro';
import renderer from 'react-test-renderer';

it('Euro with cents renders correctly', () => {
    const tree = renderer.create(<Euro>{1000}</Euro>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Euro without cents  renders correctly', () => {
    const tree = renderer.create(<Euro cents={false}>{1000}</Euro>).toJSON();
    expect(tree).toMatchSnapshot();
});
