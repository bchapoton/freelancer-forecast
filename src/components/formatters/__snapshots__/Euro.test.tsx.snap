// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Euro with bold  renders correctly 1`] = `
<span
  className="MuiBox-root css-1vg6q84"
>
  1 000 €
</span>
`;

exports[`Euro with cents renders correctly 1`] = `
<span
  className="MuiBox-root css-1l7mgse"
>
  1 000,00 €
</span>
`;

exports[`Euro without cents  renders correctly 1`] = `
<span
  className="MuiBox-root css-1l7mgse"
>
  1 000 €
</span>
`;
